### PR TITLE
add some more stubs for array functions

### PR DESCRIPTION
**Pull request by marcosh**
This adds some more stubs for array functions

### CODE REVIEW

File: src/Psalm/Internal/Stubs/CoreGenericFunctions.phpstub [hunk 1]
```diff
@@ -49,6 +49,25 @@ function array_intersect_key(array $arr, array $arr2, array ...$arr3)
...
  * @param array<TKey, TValue> $arr
+ * @param array $arr2
+ * @param array ...$arr3
+ * @param callable(TKey): int $keyCompareFunc
+ *
+ * @return array<TKey, TValue>
+ *
+ * @psalm-pure
+ */
+function array_intersect_ukey(array $arr, array $arr2, array ...$arr3, callable $keyCompareFunc): array {}
```
Comment (andrew-demb) [inline src/Psalm/Internal/Stubs/CoreGenericFunctions.phpstub:60]: Brackets mentioned twice

Comment (marcosh) [reply src/Psalm/Internal/Stubs/CoreGenericFunctions.phpstub]: thanks @andrew-demb

**Comment by muglug** [comment]
Please remove the six stubs that are already covered by return type providers in Psalm (currently `array_fill`, `array_merge`, `array_pad`, `array_pop`, `array_rand`, `array_reduce`).

**Comment by marcosh** [comment]
@muglug done. Just for the sake of understanding, are return type providers used when annotations are not enough to type correctly the function?

**Comment by muglug** [comment]
> are return type providers used when annotations are not enough to type correctly the function?

Yes, exactly. When I introduced conditional return types I removed (I think) all the return type providers that were simple enough to be implemented instead as conditional return types.

**Comment by muglug** [comment]
Thanks!

**Pull request merged**